### DISCUSSION TITLE
Collection endpoint 'method' field should only accept uppercase

### DISCUSSION THREAD

**Discussion by tyteen4a03** [payloadcms/payload#8312] [Feature Requests & Ideas]
[RFC7231](https://datatracker.ietf.org/doc/html/rfc7231#section-4.1) specifies that HTTP methods are case-sensitive, and only defines methods in uppercase, but Payload's types only accept lowercase method names.

This should be changed to only allowing uppercase for v3 release.

**Comment by akhrarovsaid** [comment]
You are correct, according to RFC7230 - RFC7321 method names must be uppercase. However, it's not actually clear to me if Payload automatically converts method names to uppercase behind the scenes - doesn't seem like it though. (I think we'd find that [here](https://github.com/payloadcms/payload/blob/v3.0.0-beta.107/packages/next/src/routes/rest/index.ts) but I may be mistaken.)

I agree, V3 should conform to existing conventions with custom endpoint methods.
